packages/core/src/proxy_configuration.ts

```diff
@@ -189,6 +265,52 @@ export class ProxyConfiguration {
         };
     }
 
+    /**
+     * Given a session identifier and a request / proxy tier, this function returns a new proxy URL based on the provided configuration options.
+     * @param _sessionId Session identifier
+     * @param options Options for the tiered proxy rotation
+     * @returns A string with a proxy URL.
+     */
+    protected _handleTieredUrl(_sessionId: string, options?: TieredProxyOptions): string {
+        if (!this.tieredProxyUrls) throw new Error('Tiered proxy URLs are not set');
+
+        if (!options || (!options?.request && options?.proxyTier === undefined)) {
+            const allProxyUrls = this.tieredProxyUrls.flat();
+            return allProxyUrls[this.nextCustomUrlIndex++ % allProxyUrls.length];
+        }
+
+        let tierPrediction = options.proxyTier!;
+
+        if (typeof tierPrediction !== 'number') {
+            tierPrediction = this.getProxyTier(options.request!)!;
+        }
+
+        return this.tieredProxyUrls![tierPrediction][this.nextCustomUrlIndex++ % this.tieredProxyUrls![tierPrediction].length];
```

Comment:
That's too dense of a oneliner for my taste - could you split it to make it more readable?